flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/data/FlinkParquetReaders.java

```diff
@@ -112,11 +116,14 @@ public ParquetValueReader<RowData> struct(Types.StructType expected, GroupType s
       List<ParquetValueReader<?>> reorderedFields = Lists.newArrayListWithExpectedSize(
           expectedFields.size());
       List<Type> types = Lists.newArrayListWithExpectedSize(expectedFields.size());
+      // Inferring MaxDefinitionLevel from parent field
+      int inferredMaxDefinitionLevel = type.getMaxDefinitionLevel(currentPath());
```

Comment:
Nit: Now that we're putting the field depth into the map `maxDefinitionLevelsById`, and we have the same check for `idToConstant.containsKey(id)`, do we need to have this fallback?

Reply:
@kbendick, we could not update the `maxDefinitionLevelsById ` if the `fieldType.getId()` is null, you could see it at L101.